src/mmw/apps/user/views.py

```diff
@@ -93,10 +93,36 @@ def itsi_auth(request):
     else:
         # User did not authenticate. Save their ITSI ID and send to /register
         request.session['itsi_id'] = itsi_user['id']
-        return redirect('/register')  # TODO Make this Backbone route
+        return redirect('/register/{username}/{first_name}/{last_name}'.format(
+            **itsi_user['extra']
+        ))
 
 
+@decorators.api_view(['POST'])
+@decorators.permission_classes((AllowAny, ))
 def itsi_register(request):
+    # Validate request
+    errors = []
+    if 'itsi_id' not in request.session:
+        errors.append("There was an error in authenticating you with ITSI")
+
+    if 'username' not in request.POST or not request.POST.get('username'):
+        errors.append("Username must be specified")
+    elif User.objects.filter(username=request.POST.get('username')).exists():
+        errors.append("Username already exists")
+
+    if 'first_name' not in request.POST or not request.POST.get('first_name'):
+        errors.append("First name must be specified")
+    if 'last_name' not in request.POST or not request.POST.get('last_name'):
+        errors.append("Last name must be specified")
+    if 'agreed' not in request.POST or not request.POST.get('agreed'):
+        errors.append("You must agree to the terms")
+
+    if len(errors) > 0:
+        response_data = {"errors": errors}
+        return Response(data=response_data,
+                        status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
If this style is too verbose, of if there is a more Pythonic way of doing this, please let me know.